src/os/desktop.zig

```diff
@@ -59,3 +59,29 @@ pub fn launchedFromDesktop() bool {
         else => @compileError("unsupported platform"),
     };
 }
+
+pub const DesktopEnvironment = enum {
+    gnome,
+    macos,
+    other,
+    windows,
+};
+
+/// Detect what desktop environment we are running under. This is mainly used on
+/// Linux to enable or disable GTK client-side decorations but there may be more
+/// uses in the future.
+pub fn desktopEnvironment() DesktopEnvironment {
+    return switch (comptime builtin.os.tag) {
+        .macos => .macos,
+        .windows => .windows,
+        .linux => de: {
+            if (@inComptime()) @compileError("Checking for the desktop environment on Linux must be done at runtime.");
```

Comment:
This is unnecessary. The functions called below won't work in comptime, the compiler will error for us.

Reply:
The call to `posix.getenv` does indeed fail, but it fails more because of an implementation detail in Zig that could change in the future rather than something more fundamental so I would prefer to be explicit about why we can't check for DE at comptime (and provide a more informative error): https://github.com/ziglang/zig/issues/4524

`std.ascii.eqlIgnoreCase` works just fine at comptime.